packages/auto-label/src/auto-label.ts

```diff
@@ -211,66 +240,175 @@ handler.autoDetectLabel = (
+    jsonArray,
+    owner,
+    repo
+  );
+
+  let autoDetectedLabelAdded: string | undefined;
+
+  const labelsOnIssue = await handler.checkExistingIssueLabels(
+    context.github,
+    owner,
+    repo,
+    issueNumber
+  );
+
+  if (!objectInJsonArray?.github_label) {
+    console.log(
+      `There was no configured match for the repo ${repo}, trying to auto-detect the right label`
+    );
+    const autoDetectedLabel = handler.autoDetectLabel(
+      jsonArray,
+      context.payload.issue.title
+    );
+    if (autoDetectedLabel) {
+      console.log(
+        `Auto-detected label ${autoDetectedLabel} for ${owner}/${repo}#${issueNumber}`
+      );
+      autoDetectedLabelAdded = await handler.maybeAddAPILabel(
+        context.github,
+        owner,
+        repo,
+        context.payload.issue.number,
+        labelsOnIssue,
+        autoDetectedLabel
+      );
+    } else {
+      console.log(
+        `Auto-detect did not add a label for ${owner}/${repo}#${issueNumber}: '${context.payload.issue.title}'`
+      );
+    }
+    return;
+  }
+
+  const colorNumber = jsonArray.findIndex(
+    (object: JSONData) => objectInJsonArray === object
+  );
+  const githubLabel = objectInJsonArray.github_label;
+
+  const alreadyExists = await handler.checkExistingLabels(
+    context.github,
+    owner,
+    repo,
+    githubLabel
+  );
+
+  if (alreadyExists === null || alreadyExists === undefined) {
+    handler.createLabel(
+      context.github,
+      owner,
+      repo,
+      githubLabel,
+      colorsData[colorNumber].color
+    );
+  } else {
+    console.log(
+      'This label already exists on the repository, will check if it also exists on the issue'
+    );
+  }
+
+  if (labelsOnIssue) {
+    const found = labelsOnIssue.find(
+      (element: {name: string}) => element.name === githubLabel
+    );
+    const cleanUpOtherLabels = labelsOnIssue.filter(
+      (element: {name: string}) =>
+        element.name.startsWith('api') &&
+        element.name !== found?.name &&
+        element.name !== autoDetectedLabelAdded
+    );
+    if (found) {
+      console.log('This label already exists on this issue');
+      return;
+    } else {
+      await handler.addLabels(context.github, owner, repo, issueNumber, [
+        `${objectInJsonArray.github_label}`,
+      ]);
+    }
+    for (const dirtyLabel of cleanUpOtherLabels) {
+      await handler.removeLabel(
+        context.github,
+        owner,
+        repo,
+        issueNumber,
+        dirtyLabel.name
+      );
+    }
+  } else {
+    await handler.addLabels(context.github, owner, repo, issueNumber, [
+      `${objectInJsonArray.github_label}`,
+    ]);
+  }
+
+  return;
+};
+
+//main function, responds to label being added
 function handler(app: Application) {
-  app.on(['issues.labeled'], async context => {
-    const owner = context.payload.repository.owner.login;
+  //nightly cron that backfills and corrects api labels
+  app.on(['schedule.repository'], async context => {
```

Comment:
Do we also have an event for when the app is installed?  Thinking that right after install is the right time to do the first creation of labels, and then the cron to keep them up to date as we go feels right.

Reply:
Added an installation event!